src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -30,7 +30,32 @@ internal JsonTypeInfo GetOrAddJsonTypeInfo(Type type)
                 InitializeCachingContext();
             }
 
-            return _cachingContext.GetOrAddJsonTypeInfo(type);
+            return GetOrAddJsonTypeInfoEx(type, configured: true, doNotThrow: false)!;
+        }
+
+        internal JsonTypeInfo? GetOrAddJsonTypeInfoEx(Type type, bool configured, bool doNotThrow)
+        {
+            if (_cachingContext == null)
+            {
+                InitializeCachingContext();
+            }
+
+            JsonTypeInfo? typeInfo = _cachingContext.GetOrAddJsonTypeInfo(type);
+
+            if (typeInfo == null)
+            {
+                if (!doNotThrow)
```

Comment:
I would just remove this and have the calling method(s) throw the appropriate exception message.

Reply:
Actually seems I over-complicated 😄